src/sentry/migrations/0374_backfill_monitor_environments.py

```diff
@@ -0,0 +1,91 @@
+# Generated by Django 2.2.28 on 2023-03-07 17:55
+from django.db import connection, migrations
+from psycopg2.extras import execute_values
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+BATCH_SIZE = 100
+DEFAULT_ENVIRONMENT_NAME = "production"
+
+UPDATE_QUERY = """
+    UPDATE sentry_monitorcheckin
+    SET monitor_environment_id = data.monitor_environment_id
+    FROM (VALUES %s) AS data (id, monitor_environment_id)
+    WHERE sentry_monitorcheckin.id = data.id"""
+
+
+def backfill_monitor_checkins(apps, schema_editor):
+    MonitorCheckIn = apps.get_model("sentry", "MonitorCheckIn")
+    MonitorEnvironment = apps.get_model("sentry", "MonitorEnvironment")
+    Environment = apps.get_model("sentry", "Environment")
+    EnvironmentProject = apps.get_model("sentry", "EnvironmentProject")
+
+    queryset = RangeQuerySetWrapperWithProgressBar(
+        MonitorCheckIn.objects.all().select_related("monitor")
+    )
```

Comment:
I'd recommend doing something like this:
```
    queryset = RangeQuerySetWrapperWithProgressBar(
        MonitorCheckIn.objects.values_list("id", "monitor_id"), result_value_getter=lambda item: item[0]
    )
 ```

We don't really need the entire monitor checkin row in memory, as far as I know you just need the id and the monitor_id for the lookup. This saves us loading a whole bunch of unnecessary data and should help speed this migration up.